llvm/test/Transforms/LoopVectorize/RISCV/csa.ll

```diff
@@ -0,0 +1,5188 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt < %s -S -mtriple riscv64 -mattr="+v" -riscv-v-vector-bits-min=256 \
+; RUN:   -passes=loop-vectorize -force-tail-folding-style=data-with-evl \
+; RUN:   -enable-csa-vectorization | FileCheck %s -check-prefix=EVL
+; RUN: opt < %s -S -mtriple riscv64 -mattr="+v" -riscv-v-vector-bits-min=256 \
+; RUN:   -passes=loop-vectorize -force-tail-folding-style=none \
+; RUN:   -enable-csa-vectorization | FileCheck %s -check-prefix=NO-EVL
+; RUN: opt < %s -S -mtriple riscv64 -mattr="+v" -riscv-v-vector-bits-min=256 \
+; RUN:   -passes=loop-vectorize -force-tail-folding-style=data \
+; RUN:   -enable-csa-vectorization | FileCheck %s -check-prefix=DATA
```

Comment:
I'm not sure if it's productive to check against tail-folding-styles. Maybe force-vector-width, force-vector-interleave instead?

Reply:
We are generating different recipes for different tail folding styles. We should test against tail folding styles since that gives us coverage for those code paths.